kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/Protocol.java

```diff
@@ -512,7 +519,20 @@ public Protocol merge(Protocol... others) {
 
   /** Check if the protocol supports the given table feature */
   public boolean supportsFeature(TableFeature feature) {
-    return getImplicitlyAndExplicitlySupportedFeatures().contains(feature);
+    if (feature.isReaderWriterFeature()) {
+      if (supportsReaderFeatures) {
+        // Note: a RW feature should also be in writerFeatures
```

Comment:
Curious: What should I learn/deduce from this comment?
- we have an `if` on is ReaderWriterFeature
- we have an `if` on supports reader
- // why does it matter if this RW should also be in writerFeatures? what's the impact / connection / consequence?

Hope my question makes sense! Basically: would a naive implementation have _also_ checked the writer features? or something like that?